### PR TITLE
synaptics-mst: Add board ID firmware offset for Cayenne and Spyder

### PR DESCRIPTION
**Pull request by hughsie**
Get the device family from firmware file, using the hardware if possible.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/synaptics-mst/fu-synaptics-mst-firmware.c [hunk 1]
```diff
@@ -59,14 +114,33 @@ fu_synaptics_mst_firmware_parse(FuFirmware *firmware,
...
 	g_autoptr(GByteArray) buf = g_byte_array_new();
 	g_autoptr(GBytes) blob = NULL;
-
+	guint16 addr;
+
+	switch (self->family) {
+	case FU_SYNAPTICS_MST_FAMILY_TESLA:
+	case FU_SYNAPTICS_MST_FAMILY_LEAF:
+	case FU_SYNAPTICS_MST_FAMILY_PANAMERA:
+		addr = ADDR_CUSTOMER_ID;
```
Comment (superm1) [inline plugins/synaptics-mst/fu-synaptics-mst-firmware.c:126]: Maybe addr_customer_id_tesla to follow the naming convention used for Cayenne.

File: plugins/synaptics-mst/fu-synaptics-mst-firmware.c [hunk 2]
```diff
@@ -59,14 +114,33 @@ fu_synaptics_mst_firmware_parse(FuFirmware *firmware,
...
+		break;
+	case FU_SYNAPTICS_MST_FAMILY_CAYENNE:
+	case FU_SYNAPTICS_MST_FAMILY_SPYDER:
+		addr = ADDR_CUSTOMER_ID_CAYENNE;
+		break;
+	default:
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "Unsupported chip family");
```
Comment (superm1) [inline plugins/synaptics-mst/fu-synaptics-mst-firmware.c:136]: Can you put the family in the error or set a literal error instead?

File: plugins/synaptics-mst/fu-synaptics-mst.rs [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
-    Unknown,
-    Tesla,
-    Leaf,
-    Panamera,
-    Cayenne,
-    Spyder,
+    Unknown = 0xFF,
+    Tesla = 0,
+    Leaf = 1,
+    Panamera = 2,
```
Comment (hughsie) [inline plugins/synaptics-mst/fu-synaptics-mst.rs:9]: @CloudExt I have no idea if this is true -- I wondered if generation 1 should really be `TeslaLeaf` or something.

Comment (CloudExt) [reply plugins/synaptics-mst/fu-synaptics-mst.rs:9]: That's correct.

**Review by CloudExt — APPROVED** [review]

File: plugins/synaptics-mst/fu-synaptics-mst-firmware.c (hunk 2 above)
Comment (hughsie) [reply plugins/synaptics-mst/fu-synaptics-mst-firmware.c:136]: Good idea, fixed.

File: plugins/synaptics-mst/fu-synaptics-mst.rs (hunk 3 above)
Comment (hughsie) [reply plugins/synaptics-mst/fu-synaptics-mst.rs:9]: @CloudExt as in the code I wrote is correct? or TeslaLeaf? Thanks!

File: plugins/synaptics-mst/fu-synaptics-mst-firmware.c (hunk 1 above)
Comment (hughsie) [reply plugins/synaptics-mst/fu-synaptics-mst-firmware.c:126]: Done

File: plugins/synaptics-mst/fu-synaptics-mst.rs (hunk 3 above)
Comment (CloudExt) [reply plugins/synaptics-mst/fu-synaptics-mst.rs:9]: Your code is correct, we have two old product lines Tesla and Leaf.

**Review by CloudExt — APPROVED** [review]

**Pull request merged**